extensions/transfer/transfer-core/src/main/java/com/microsoft/dagx/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -120,9 +123,25 @@ private void run() {
         }
     }
 
+    private int pollForCompletion() {
```

Comment:
Two things we will need to think about:

1. Eventually we need to formalize the notion of finite and non-finite transfer processes. Non-finite ones should not be polled. 
2. We need to handle the "starvation" scenario where there are too many processes and some are not checked due to batching. This could potentially be handled by updating the state timestamp on each process after each check. This will put a recently checked process to the back of the "queue".